packages/plugin_platform_interface/lib/plugin_platform_interface.dart

```diff
@@ -57,6 +57,7 @@ abstract class PlatformInterface {
   /// This is implemented as a static method so that it cannot be overridden
   /// with `noSuchMethod`.
   static void verifyToken(PlatformInterface instance, Object token) {
+    assert(token != const Object());
```

Comment:
Per discussion in the issue, we can't add this without making a breaking version change. What we can do though is soft-deprecate this (i.e., put a comment that it shouldn't be used any more, but not an annotation since that would break our own builds) and make a new method that has this assertion then calls `verifyToken`.

(We can then update all the 1P plugins to call the new method, then make it a real deprecation and move the implementation to a shared private method so we're not calling the deprecated version.)

Reply:
I've implemented your suggested changes, PTAL